src/createAnimatedComponent.tsx

```diff
@@ -601,12 +604,11 @@ export default function createAnimatedComponent(
         >,
       setLocalRef: (ref) => {
         // TODO update config
-
         const tag = IS_WEB
           ? (ref as HTMLElement)
           : findNodeHandle(ref as Component);
-
         const { layout, entering, exiting, sharedTransitionTag } = this.props;
+        const skipEntering = !!this.context?.current;
```

Comment:
small think, but we can move the declaration of `skipEntering` right before `if (entering && !skipEntering) {`